test/e2e/lib/gutenberg/gutenberg-editor-component.js

```diff
@@ -676,4 +676,23 @@ export default class GutenbergEditorComponent extends AsyncBaseContainer {
 			By.css( '.edit-post-header .table-of-contents button' )
 		);
 	}
+
+	async toggleNavigationSidebar() {
+		const isOpen = await driverHelper.isElementLocated(
+			this.driver,
+			By.css( '.wpcom-block-editor-nav-sidebar-nav-sidebar__click-guard' )
+		);
+
+		if ( isOpen ) {
+			await driverHelper.clickWhenClickable(
+				this.driver,
+				By.css( '.wpcom-block-editor-nav-sidebar-nav-sidebar__dismiss-sidebar-button' )
+			);
+		} else {
+			await driverHelper.clickWhenClickable(
+				this.driver,
+				By.css( '.wpcom-block-editor-nav-sidebar-toggle-sidebar-button__button' )
```

Comment:
I think this selector works for toggling regardless of open/close state.

Reply:
It doesn't work. The button is not visible so Selenium can't click on it. That's why we need to use the button in the sidebar.

Reply:
I think if you let CSS select the first available element it should toggle the sidebar alright:

```js
const dismissSidebarButtonSelector =  '.wpcom-block-editor-nav-sidebar-nav-sidebar__dismiss-sidebar-button';
const toggleSidebarButtonSelector = '.wpcom-block-editor-nav-sidebar-toggle-sidebar-button__button';

async toggleNavigationSidebar() {
	return await driverHelper.clickWhenClickable(
		this.driver,
		By.css( toggleSidebarButtonSelector + ', ' + dismissSidebarButtonSelector );
	);
}
```

...assuming those selectors are correct 😉

Reply:
Just checked those out in Gutenberg, and it seems we can use aria attributes as selectors. It's generally a better practice to use user-facing attributes as selectors as they're less likely to change and break the test. Notice also the rename to `toggleBlockEditorSidebar` as it seems to be the UI component name. Here's the updated version of the above toggle method:

```js
async toggleBlockEditorSidebar() {
	const dismissSidebarButtonSelector =  'button[aria-label="Block editor sidebar"][aria-expanded="false"]';
	const toggleSidebarButtonSelector = 'button[aria-label="Close block editor sidebar"]';

	return await driverHelper.clickWhenClickable(
		this.driver,
		By.css( toggleSidebarButtonSelector + ', ' + dismissSidebarButtonSelector );
	);
}

```